packages/flutter_tools/lib/src/reporting/unified_analytics.dart

```diff
@@ -64,3 +67,41 @@ int? getMaxRss(ProcessInfo processInfo) {
   }
   return null;
 }
+
+/// Get the analytics related informatin for null safety analysis within
+/// a flutter package for reporting.
+Map<String, Object>? getNullSafetyAnalysisInfo({
+  required PackageConfig packageConfig,
+  required NullSafetyMode nullSafetyMode,
+  required String currentPackage,
+}) {
+  if (packageConfig.packages.isEmpty) {
+    return null;
+  }
+  int migrated = 0;
+  LanguageVersion? languageVersion;
+  for (final Package package in packageConfig.packages) {
+    final LanguageVersion? packageLanguageVersion = package.languageVersion;
+    if (package.name == currentPackage) {
+      languageVersion = packageLanguageVersion;
+    }
+    // TODO: (eliasyishak) this logic is copied from "lib/src/reporting/events.dart"
+    //  but is this correct, this would fail if comparing 3.1 to 2.12
+    if (packageLanguageVersion != null &&
+        packageLanguageVersion.major >= nullSafeVersion.major &&
+        packageLanguageVersion.minor >= nullSafeVersion.minor) {
```

Comment:
Is this correct? This logic for calculating total number of migrated packages was taken from [here](https://github.com/eliasyishak/flutter/blob/ga4-migrate-null-safety-analysis-event/packages/flutter_tools/lib/src/reporting/events.dart#L287-L289)

I think we should be able to just compare two `LanguageVersion` classes together instead of comparing the major and minor versions